### PR TITLE
Bugfix/nullable transaction return

### LINKED ISSUE #557: Transaction: a nullable return type becomes not nullable in the method's implementation
Step to reproduce the issue:
- add this simple method to the TaskDao class in the official [example](https://github.com/vitusortner/floor/blob/develop/example/lib/task_dao.dart)

  ```dart
  @transaction
   Future<Task?> test(int id) {
     return findTaskById(id);
   }
  ```
- run `flutter pub run build_runner build`

Generated code:
```dart
  @override
  Future<Task> test(int id) async {
    if (database is sqflite.Transaction) {
      return super.test(id);
    } else {
      return (database as sqflite.Database)
          .transaction<Task>((transaction) async {
        final transactionDatabase = _$FlutterDatabase(changeListener)
          ..database = transaction;
        return transactionDatabase.taskDao.test(id);
      });
    }
  }
```

This code doesn't compile because the method is declared with a not nullable return type, and also `sqflite.Database.transaction<Task>(..)` (in the else branch) is invoked with a not nullable parametric type

### PR DESCRIPTION
**Pull request by mqus**
Fixes #557 and maybe adresses #558

### CODE REVIEW

File: floor_generator/lib/writer/transaction_method_writer.dart [hunk 1]
```diff
@@ -14,7 +14,7 @@ class TransactionMethodWriter implements Writer {
     return Method((builder) => builder
       ..annotations.add(overrideAnnotationExpression)
       ..returns = refer(method.returnType.getDisplayString(
-        withNullability: false,
+        withNullability: true,
```
Comment (vitusortner) [inline floor_generator/lib/writer/transaction_method_writer.dart:17]: ❓ Do we also get support for the case described in https://github.com/vitusortner/floor/issues/557 where we don't return a nullable `Future` but a `Future` wrapping a nullable type (e.g `Future<Task?>`)? If so, could you add another test to verify the behavior?

**Review by vitusortner — COMMENTED** [review]
Thanks for the quick fix!

File: floor_generator/lib/writer/transaction_method_writer.dart (hunk 1 above)
Comment (mqus) [reply floor_generator/lib/writer/transaction_method_writer.dart:17]: I think we had that support already, but I'll add a test.

**Review by vitusortner — APPROVED** [review]

**Comment by xThalia** [comment]
I've tried to return Future<Object?> from transaction but it still causes an error in generated db file. Will there be support for return Future with nullable object?

**Pull request merged**